client/src/util/lang.ts

```diff
@@ -13,7 +13,7 @@ export function compactObject<T>(obj: Record<string, T | null | undefined>): Rec
 export type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
 
 export function isObject(value: any): value is object {
-  return value !== undefined && value !== null && typeof value === 'object';
+  return value !== undefined && value !== null && typeof value === 'object' && !Array.isArray(value);
```

Comment:
Nitpick. Maybe we can do this expression sorter:

```suggestion
  return value != null && typeof value === 'object' && !Array.isArray(value);
```

Reply:
I would rather do:

```suggestion
  return Boolean(value) && typeof value === 'object' && !Array.isArray(value);
```